libeos-payg/boottime-source.c

```diff
@@ -113,11 +117,19 @@ epg_boottime_source_new (guint interval_ms)
 
   its.it_value = its.it_interval;
 
+  /* Set the GError if timerfd_create() fails because it could be e.g. ENFILE
+   * which we should handle gracefully */
   fd = timerfd_create (CLOCK_BOOTTIME, TFD_CLOEXEC | TFD_NONBLOCK);
   if (G_UNLIKELY (fd < 0))
-    g_error ("timerfd_create (CLOCK_BOOTTIME, TFD_CLOEXEC) failed: %s",
-             g_strerror (errno));
+    {
+      g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
```

Comment:
Use `g_io_error_from_errno()`. I wish `G_IO_ERROR_FAILED` had never been invented; people use it too much when more specific errors are available.